apps/webservice/src/app/[workspaceSlug]/settings/(settings)/workspace/members/WorkspaceMembersTable.tsx

```diff
@@ -41,32 +42,35 @@ interface Member {
 
 const InviteLinkSection: React.FC<{
   sessionMember?: Member;
-}> = ({ sessionMember }) => {
-  const inviteLink = api.invite.workspace.link.byWorkspaceMemberId.useQuery(
-    sessionMember?.workspace_member.id ?? "",
-    { enabled: sessionMember != null },
-  );
-  const { mutateAsync } = api.invite.workspace.link.create.useMutation();
+  workspaceSlug: string;
+  inviteLink?: string;
+}> = ({ sessionMember, workspaceSlug, inviteLink }) => {
+  const workspace = api.workspace.bySlug.useQuery(workspaceSlug);
   const utils = api.useUtils();
+  const { mutateAsync } = api.invite.workspace.link.create.useMutation({
+    onSuccess: () =>
+      utils.invite.workspace.link.byWorkspaceMemberId.invalidate(),
+  });
   const [clickedCopy, setClickedCopy] = useState(false);
 
-  useEffect(() => {
-    if (inviteLink.isSuccess && inviteLink.data == null)
-      mutateAsync(sessionMember?.workspace_member.id ?? "").then(() =>
-        utils.invite.workspace.link.byWorkspaceMemberId.invalidate(
-          sessionMember?.workspace_member.id ?? "",
-        ),
-      );
-  }, [mutateAsync, inviteLink, sessionMember?.workspace_member.id, utils]);
-
-  const link = `${env.NEXT_PUBLIC_BASE_URL}/join/${inviteLink.data?.token ?? ""}`;
+  const [token] = useState(inviteLink ?? v4());
```

Comment:
I had to keep it like this to prevent a ui issue where when the db upsert happendd and the link got invalidate, it would show a new v4()